oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthCodeHandler.java

```diff
@@ -187,4 +211,44 @@ public APIGatewayProxyResponseEvent handleRequest(
                                                     authenticationResponse.toURI().toString()));
                         });
     }
+
+    private void updateClientSubjects(
+            UserProfile userProfile, AuthenticationRequest authenticationRequest) {
+        String clientId = authenticationRequest.getClientID().getValue();
+        ClientRegistry clientRegistry = dynamoClientService.getClient(clientId).get();
+
+        String sector =
+                clientRegistry.getSectorIdentifierUri() != null
+                        ? clientRegistry.getSectorIdentifierUri()
+                        : null;
+
+        if (clientRegistry.getSubjectType().equalsIgnoreCase("pairwise")) {
+            if (userProfile.getClientSubjects() != null) {
+                if (!clientSubjectsContainsClientId(userProfile.getClientSubjects(), clientId)) {
+                    userProfile
+                            .getClientSubjects()
+                            .add(new ClientSubject(clientId, new Subject(), sector));
+                } else {
+                    userProfile.setClientSubjects(
+                            List.of(new ClientSubject(clientId, new Subject(), sector)));
```

Comment:
I don't we should be doing `new Subject()` here, we should be retrieving the existing one from the list.

Reply:
In fact, I don't think we need this `else` at all!?